lecture_3/03_project/calculator/internal/engine/handler/handler.go

```diff
@@ -0,0 +1,109 @@
+package handler
+
+import (
+	"context"
+	domainmodels "github.com/koines/code-cadets-2021/lecture_3/03_project/calculator/internal/domain/models"
+	rabbitmqmodels "github.com/koines/code-cadets-2021/lecture_3/03_project/calculator/internal/infrastructure/rabbitmq/models"
+	"log"
+)
+
+// Handler handles bets received and bets calculated.
+type Handler struct {
+	betCalcRepository BetCalcRepository
+}
+
+// New creates and returns a new Handler.
+func New(betCalcRepository BetCalcRepository) *Handler {
+	return &Handler{
+		betCalcRepository: betCalcRepository,
+	}
+}
+
+// HandleBets handles bets received.
+func (h *Handler) HandleBets(
+	ctx context.Context,
+	bets <-chan rabbitmqmodels.Bet,
+) {
+
+	go func() {
+		for bet := range bets {
+			log.Println("Processing bet received, betId:", bet.Id)
+
+			// Calculate the domain bet based on the incoming bet received.
+			domainBet := domainmodels.Bet{
+				Id:                   bet.Id,
+				SelectionId:          bet.SelectionId,
+				SelectionCoefficient: bet.SelectionCoefficient,
+				Payment:              bet.Payment,
+			}
+
+			// Insert the domain bet into the repository.
+			err := h.betCalcRepository.InsertBet(ctx, domainBet)
+			if err != nil {
+				log.Println("Failed to insert bet, error: ", err)
+				continue
+			}
+
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				continue
+			}
```

Comment:
how do you stop the infinite loop here which happens when the bet is calculated it is then produced to the `bets` topic again.

Somehow you need to skip those messages